internal/core/encoding.go

```diff
@@ -288,3 +288,81 @@ func EncodeIndexDataStoreKey(key *IndexDataStoreKey) []byte {
 
 	return b
 }
+
+func DecodeDataStoreKey(data []byte) (DataStoreKey, error) {
```

Comment:
suggestion: A line documenting the decoding type

Reply:
Will add

- [ ] doc func